client/my-sites/domains/domain-management/settings/cards/registered-domain-details.tsx

```diff
@@ -88,7 +88,7 @@ const RegisteredDomainDetails = ( {
 		return (
 			<>
 				<AutoRenewToggle
-					planName={ selectedSite.plan.product_name_short }
+					planName={ selectedSite.plan?.product_name_short }
```

Comment:
`AutoRenewToggle` already accepts an undefined `planName`; this will just prevent a fatal error if it's undefined.